articles/azure-functions/functions-reference-powershell.md

```diff
@@ -0,0 +1,424 @@
+
+The following example sets the threshold to enable verbose logging for all functions, but sets the threshold to enable debug logging for a single function:
+
+```json
+{
+    "logging": {
+        "logLevel": {
+            "Function.MyFunction": "Debug",
+            "default": "Trace"
+        }
+    }
+}  
+```
+
+For more information, see [host.json reference](functions-host-json.md).
+
+## HTTP triggers and bindings
+
+HTTP and webhook triggers and HTTP output bindings use request and response objects to represent the HTTP messaging.  
+
+### Request object
+
+The Request object that's passed into the script comes from a type called `HttpRequestContext` which has the following properties:
+
+| Property  | Description                                                    |
+|-----------|----------------------------------------------------------------|
+| _Body_    | An object that contains the body of the request.               |
+| _Headers_ | A dictionary that contains the request headers.                |
+| _Method_  | The HTTP method of the request.                                |
+| _Params_  | An object that contains the routing parameters of the request. |
+| _Query_   | An object that contains the query parameters.                  |
+| _Url_     | The URL of the request.                                        |
+
+### Response object
+
+The Response object that you should send back comes from a type called `HttpResponseContext` which has the following properties:
+
+| Property      | Description                                                 |
+|---------------|-------------------------------------------------------------|
+| _Body_        | An object that contains the body of the response.           |
+| _ContentType_ | A short hand for setting the content type for the response. |
+| _Headers_     | An object that contains the response headers.               |
+| _StatusCode_  | The HTTP status code of the response.                       |
+
+### Accessing the request and response
+
+When you work with HTTP triggers, you can access the HTTP request the same way you would with any other input binding. It's in the `param` block. 
+
+In order to send back a response, you need to send back an `HttpResponseContext` object. Here's an example:
+
+`function.json`
+```json
+{
+  "bindings": [
+    {
+      "type": "httpTrigger",
+      "direction": "in",
+      "authLevel": "anonymous"
+    },
+    {
+      "type": "http",
+      "direction": "out"
+    }
+  ]
+}
+```
+
+`run.ps1`
+```powershell
+param($req, $TriggerMetadata)
+
+$name = $req.Query.Name
+
+Push-OutputBinding -Name res -Value ([HttpResponseContext]@{
+    StatusCode = $status
```

Comment:
Seems like the user shouldn't have to cast the hashtable, we should do that internally for them.  Also `$status` isn't defined, so this isn't a complete working example.  Perhaps just set to:

```suggestion
    StatusCode = [System.Net.HttpStatusCode]::OK
```

Reply:
good catch. I've opened an issue on the casting here https://github.com/Azure/azure-functions-powershell-worker/issues/93